client/pruning/main.go

```diff
@@ -0,0 +1,125 @@
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			vp := viper.New()
+
+			// Bind flags to the Context's Viper so we can get pruning options.
+			if err := vp.BindPFlags(cmd.Flags()); err != nil {
+				return err
+			}
+			pruningOptions, err := server.GetPruningOptionsFromFlags(vp)
+			if err != nil {
+				return err
+			}
+
+			home := vp.GetString(flags.FlagHome)
+			db, err := openDB(home, server.GetAppDBBackend(vp))
+			if err != nil {
+				return err
+			}
+
+			logger := log.NewTMLogger(log.NewSyncWriter(os.Stdout))
+			app := appCreator(logger, db, nil, vp)
+			cms := app.CommitMultiStore()
+
+			if rootMultiStore, ok := cms.(*rootmulti.Store); ok {
+				latestHeight := rootmulti.GetLatestVersion(db)
+				// valid heights should be greater than 0.
+				if latestHeight <= 0 {
+					return fmt.Errorf("the database has no valid heights to prune, the latest height: %v", latestHeight)
+				}
+
+				shouldBePruned := shouldBePrunedValidator(pruningOptions, latestHeight)
+
+				var pruningHeights []int64
+				for height := int64(1); height < latestHeight; height++ {
+					if shouldBePruned(height) {
+						pruningHeights = append(pruningHeights, height)
+					}
+				}
+				if len(pruningHeights) == 0 {
+					fmt.Printf("no heights to prune\n")
+					return nil
+				}
+				fmt.Printf(
+					"pruning heights start from %v, end at %v, with the interval of %v\n",
+					pruningHeights[0],
+					pruningHeights[len(pruningHeights)-1],
+					pruningOptions.Interval,
+				)
+
+				err = rootMultiStore.PruneStores(false, pruningHeights)
+				if err == nil {
+					fmt.Printf("successfully pruned the application root multi stores\n")
+				}
+				return err
+			}
+
+			return fmt.Errorf("currently only support the pruning of rootmulti.Store type")
+		},
+	}
+
+	cmd.Flags().String(flags.FlagHome, "", "The database home directory")
+	cmd.Flags().String(server.FlagPruning, pruningtypes.PruningOptionDefault, "Pruning strategy (default|nothing|everything|custom)")
+	cmd.Flags().Uint64(server.FlagPruningKeepRecent, 0, "Number of recent heights to keep on disk (ignored if pruning is not 'custom')")
+	cmd.Flags().Uint64(server.FlagPruningInterval, 0, "Height interval at which pruned heights are removed from disk (ignored if pruning is not 'custom')")
+	cmd.Flags().String(FlagAppDBBackend, "", "The type of database for application and snapshots databases")
+
+	return cmd
+}
+
+func openDB(rootDir string, backendType dbm.BackendType) (dbm.DB, error) {
+	dataDir := filepath.Join(rootDir, "data")
+	return dbm.NewDB("application", backendType, dataDir)
+}
+
+func shouldBePrunedValidator(options pruningtypes.PruningOptions, latestHeight int64) func(int64) bool {
+	return func(height int64) bool {
+		if height < latestHeight-int64(options.KeepRecent) {
+			return options.Interval > 0 && options.GetPruningStrategy() != pruningtypes.PruningNothing && height%int64(options.Interval) == 0
```

Comment:
it seems `options.GetPruningStrategy() != pruningtypes.PruningNothing` is not necessary?

Reply:
yes, removed this expression